Sources/NIOCore/ByteBuffer-hex.swift

```diff
@@ -14,6 +14,26 @@
 
 extension ByteBuffer {
 
+    /// Create a fresh `ByteBuffer` containing the `bytes` decoded from  the string representation of `hexEncodedBytes`.
+    ///
+    /// This will allocate a new `ByteBuffer` with enough space to fit the hex decoded `bytes` and potentially some extra space
+    /// using the default allocator.
+    ///
+    /// - info: If you have access to a `Channel`, `ChannelHandlerContext`, or `ByteBufferAllocator` we
+    ///         recommend using `channel.allocator.buffer(integer:)`. Or if you want to write multiple items into the
+    ///         buffer use `channel.allocator.buffer(capacity: ...)` to allocate a `ByteBuffer` of the right
+    ///         size followed by a `writeHexEncodedBytes` instead of using this method. This allows SwiftNIO to do
+    ///         accounting and optimisations of resources acquired for operations on a given `Channel` in the future.
+    init(hexEncodedBytes string: String) {
+        let hexPlainDecodedBytes = string.hexPlainDecodedBytes
```

Comment:
Rather than use this, we can create the buffer `var` and `writeHexEncodedString` into it.

Reply:
Can I challenge this? It doesnt seem like the pattern.
![Screenshot 2024-08-13 at 11 58 16 am](https://github.com/user-attachments/assets/795154d4-f99b-4b81-9f2f-2490a1bdaebc)

Would this suffice or do you want me to create that var? 
![Screenshot 2024-08-13 at 11 58 32 am](https://github.com/user-attachments/assets/19bff962-9625-443e-bf67-3d5a42ed47ad)

Reply:
Is it because I am calling that extra computer variable inside the initialiser that makes this not part of the original pattern and needings something new.

Reply:
Made the changes, but I think my challenge still stands.